shared/js/project.js

```diff
@@ -190,7 +200,17 @@ module.exports = {
 
     current = data;
     if (isNewChannel) {
-      location.href = current.level + '#' + current.id + '/edit';
+      // We have a new channel, meaning either we had no channel before, or
+      // we've changed channels.
+
+      if (location.hash || !window.history.pushState) {
+        // We're using a hash route or don't support replace state. Use our hash
+        // based route to ensure we don't have a page load.
+        location.href = current.level + '#' + current.id + '/edit';
+      } else {
+        window.history.pushState(null, document.title,
```

Comment:
http://diveintohtml5.info/history.html

Calling history.pushState will immediately change the URL in the browser’s location bar. So is that the end of the illusion? Well, not quite. We still need to talk about what happens when the user presses the all-important back button.

Normally when the user navigates to a new page (with a full page refresh), the browser pushes the new URL onto its history stack and downloads and draws the new page. When the user presses the back button, the browser pops one page off its history stack and redraws the previous page. But what happens now that you’ve short-circuited this navigation to avoid a full page refresh? Well, you’ve faked “moving forward” to a new URL, so now you also need to fake “moving backward” to the previous URL. And the key to faking “moving backwards” is the popstate event.